QRCoder/PdfByteQRCode.cs

```diff
@@ -55,120 +55,105 @@ public byte[] GetGraphic(int pixelsPerModule, string darkColorHtmlHex, string li
 
         //Create PDF document
         using var stream = new MemoryStream();
-#if NETFRAMEWORK
-        var writer = new StreamWriter(stream, System.Text.Encoding.ASCII);
-#elif NET6_0_OR_GREATER
-        var writer = new StreamWriter(stream, System.Text.Encoding.ASCII, leaveOpen: true);
-#else
-        var writer = new StreamWriter(stream, System.Text.Encoding.ASCII, 1024, true);
-#endif
-        try
+        using var writer = new StreamWriter(stream, Encoding.ASCII, 1024, true);
+        var xrefs = new List<long>();
+
+        // PDF header - declares PDF version 1.5
+        writer.Write("%PDF-1.5\r\n");
+        writer.Flush();
+
+        // Binary comment - ensures PDF is treated as binary file (prevents text mode corruption)
+        stream.Write(_pdfBinaryComment, 0, _pdfBinaryComment.Length);
+        writer.WriteLine();
+
+        writer.Flush();
```

Comment:
Is flushing needed? We write to a memory stream, so I don't see real benefit from it.

Same on the other flushes below.

Reply:
It's the text writer that requires flushing, not the memory stream.  Without flushing, the position may be inaccurate.  Now since we are using ASCII encoding, there is (probably) no reason for the encoder to cache characters, but I don't know that it says anywhere that the ASCII encoder doesn't require flushing.  So the encoder flushes before reading the position property.